charts/DataTableTransform.ts

```diff
@@ -108,15 +110,65 @@ function getHeaderUnit(unit: string) {
 export class DataTableTransform extends ChartTransform {
     chart: ChartConfig
 
-    @computed get isValidConfig(): boolean {
-        return true
-    }
-
     constructor(chart: ChartConfig) {
         super(chart)
         this.chart = chart
     }
 
+    @computed private get loadedWithData(): boolean {
+        return this.dimensions.length > 0
+    }
+
+    private readonly AUTO_SELECTION_THRESHOLD_PERCENTAGE: number = 0.5
+
+    /**
+     * If the user or the editor hasn't specified a start, auto-select a start year
+     *  where AUTO_SELECTION_THRESHOLD_PERCENTAGE of the entities have values.
+     */
+    @computed get autoSelectedStartYear(): number | undefined {
```

Comment:
This function looks like it would benefit greatly from unit tests to me (not only for preventing regressions, but also would probably point to a more general purpose query API on the table class), but I guess as it is now it is very easy to swap out (since it's just a method returning a number), so the impact of a bug in it wouldn't be too bad. I'd make a "todo" note somewhere to add tests :)

Reply:
Will add a to-do, I agree!